src/web-auth/web-message-handler.js

```diff
@@ -22,7 +22,8 @@ function runWebMessageFlow(authorizeUrl, options, callback) {
     timeoutCallback: function() {
       callback({
         error: 'timeout',
-        error_description: 'Timeout during executing web_message communication'
+        error_description: 'Timeout during executing web_message communication',
+        state: options.state
```

Comment:
I find odd having an error object having "extra values" just for the sake of your code simplicity. It's easy to forget to filter the attributes returned to the user later.

Reply:
the server also sends the same object `{error, error_description, state}`. I'm just replicating what the server does